src/transformers/image_transforms.py

```diff
@@ -110,10 +110,11 @@ def rescale(
     if not isinstance(image, np.ndarray):
         raise ValueError(f"Input image must be of type np.ndarray, got {type(image)}")
 
+    image = image.astype(np.float32)
```

Comment:
Image should be cast before scaling to ensure precision isn't lost when rescaling with a float value. 

This has the potential to slightly change the resulting pixel values for models. To check this doesn't affect our current models' outputs I ran integration tests for swin, vit, detr, clip and resnet.